src/charm.py

```diff
@@ -130,6 +152,24 @@ def _get_kratos_endpoint_info(self) -> Optional[str]:
 
         return kratos_public_url
 
+    def _update_login_ui_endpoint_relation_data(self, event: RelationEvent) -> None:
+        endpoint = (self.ingress.url if self.ingress.is_ready() else "",)
+
+        logger.info(f"Sending login ui endpoint info: endpoint - {endpoint[0]}")
+
+        self.endpoint_provider.send_endpoints_relation_data(endpoint[0])
+
+    def _get_hydra_endpoint_info(self) -> str:
```

Comment:
Shouldn't we return Optional[str]? If the `HydraEndpointsRelationDataMissingError` is triggered we return an empty string right now.

Reply:
updated function